hedera-node/hedera-mono-service/src/main/java/com/hedera/services/fees/PricesAndFeesImpl.java

```diff
@@ -13,32 +13,34 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.hedera.services.contracts.execution;
+package com.hedera.services.fees;
 
+import static com.hederahashgraph.fee.FeeBuilder.FEE_DIVISOR_FACTOR;
 import static com.hederahashgraph.fee.FeeBuilder.getTinybarsFromTinyCents;
 
 import com.hedera.services.context.TransactionContext;
 import com.hedera.services.evm.contracts.execution.PricesAndFeesProvider;
-import com.hedera.services.fees.FeeMultiplierSource;
-import com.hedera.services.fees.HbarCentExchange;
 import com.hedera.services.fees.calculation.UsagePricesProvider;
+import com.hedera.services.utils.MiscUtils;
+import com.hedera.services.utils.accessors.TxnAccessor;
+import com.hederahashgraph.api.proto.java.ExchangeRate;
 import com.hederahashgraph.api.proto.java.FeeComponents;
+import com.hederahashgraph.api.proto.java.FeeData;
 import com.hederahashgraph.api.proto.java.HederaFunctionality;
 import com.hederahashgraph.api.proto.java.Timestamp;
 import java.time.Instant;
 import java.util.function.ToLongFunction;
 import javax.inject.Inject;
-import javax.inject.Singleton;
 
-@Singleton
-public class LivePricesSource implements PricesAndFeesProvider {
+public class PricesAndFeesImpl implements PricesAndFeesProvider {
```

Comment:
We shouldn't substitute `LivePricesSource` with `PricesAndFeesImpl` but inject `LivePricesSource` in `PricesAndFeesImpl` and have a separate stand-alone implementation for it.